src/pages/index.js

```diff
@@ -124,23 +161,109 @@ function handleEscapeKey(evt) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editProfileModalNameInput.value;
-  profileDescription.textContent = editProfileModalDescriptionInput.value;
-  closeModal(editProfileModal);
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editProfileModalNameInput.value,
+      about: editProfileModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+
   evt.target.reset();
```

Comment:
` evt.target.reset();` should be in line 173 in `.then`

You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.

**Please, correct it everywhere in the code**